internal/app/singularity/pull.go

```diff
@@ -68,7 +68,13 @@ func LibraryPull(imgCache *cache.Handle, name, fullURI, libraryURI, keyServerURL
 		imageName := uri.GetName("library://" + imageRef)
 		imagePath := imgCache.LibraryImage(libraryImage.Hash, imageName)
 		exists, err := imgCache.LibraryImageExists(libraryImage.Hash, imageName)
-		if err != nil {
+		if err == cache.ErrBadChecksum {
+			sylog.Warningf("Removing cached image: %s: cache could be corrupted", imagePath)
```

Comment:
Why cache can be corrupted in this case? I mean, you remove image and removal is pretty transparent to the user.

Reply:
imagePath is the path to the cache, in the case of a download failure (and also fails to remove the non-complete downloaded cache), this will remove the corrupted cache, then redownload it.

Reply:
The actual output image is not copied from cache until the download is complete.

Reply:
> cache can be corrupted _only_ when malicious image removal fails

Do you mean when the `go interruptCleanup(imagePath)` catches a interrupt?

Cache can also be corrupted if the download process fails, and `go interruptCleanup(imagePath)` was unable to cache it.